test/c_api/disjoint_pool.c

```diff
@@ -24,6 +24,28 @@ void test_disjoint_pool_default_params() {
     umfMemoryProviderDestroy(provider);
 }
 
+void test_disjoint_pool_shared_limits() {
+    umf_memory_provider_handle_t provider = nullProviderCreate();
+    enum umf_result_t retp;
+    umf_memory_pool_handle_t pool = NULL;
+    struct umf_disjoint_pool_params params = umfDisjointPoolParamsDefault();
+
+    struct umf_disjoint_pool_shared_limits *limits =
+        umfDisjointPoolSharedLimitsCreate(1024);
+    params.SharedLimits = limits;
+
+    retp = umfPoolCreate(&UMF_DISJOINT_POOL_OPS, provider, &params, &pool);
+
+    // TODO: use asserts
```

Comment:
Are you going to address this TODO in this PR?

Reply:
Not in this one. I've created a task in Jira to add some helpers for C (asserts, etc.). Once this is done, we can address this.

Reply:
OK